packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -164,14 +165,22 @@ public void onLayoutChange(
   }
 
   @Override
-  @ReactProp(
-      name = ViewProps.BACKGROUND_COLOR,
-      defaultInt = Color.TRANSPARENT,
-      customType = "Color")
   public void setBackgroundColor(@NonNull T view, int backgroundColor) {
     view.setBackgroundColor(backgroundColor);
   }
 
+  @ReactProp(
+    name = ViewProps.BACKGROUND_COLOR,
+    defaultInt = Color.TRANSPARENT,
+    customType = "Color")
+  public void setBackgroundColor(@NonNull T view, long backgroundColor) {
+    if (view instanceof ReactViewGroup) {
+      ((ReactViewGroup) view).setBackgroundColor(backgroundColor);
+    } else {
```

Comment:
Hi @ryanlntn, using `ReactViewGroups` creates a circular dependencies in the internal CI, we have some difference in how modules are structured internally.

Can you refresh my memory and remind me why you need to cast the `view` to `ReactViewGroup`? Can't we use the same approach we use below for regular views?

Reply:
@cipolleschi Since I updated this to remove the use of reflection we need to cast to a type that supports `setBackgroundColor(long color)` or it won't compile. Regular views only support `setBackgroundColor(int color)`. Would it be better to create a new subclass of View to extend for `T`?

Reply:
@ryanlntn  We can't do in this way, unfortunately. Due to the circular dependencies, it will not work internally. And a subclass will give us no guarantees that it will be instantiated properly. I think that we need an interface.

The best way to move forward is probably:
1. in the `uimanager` folder, let's create an Interface:
```java
public interface ReactWideGamutView extends View {
  public void setBackgroundColor(long color);
  // other methods required by wide gamut
}
```
2. Make the `ReactViewGroup` implement the `ReactWideGamutView` interface
3. Replace the casting to `ReactViewGroup` with a check like:
```java
if (view instanceof ReactWideGamutView) {
      ((ReactWideGamutView) view).setBackgroundColor(backgroundColor);
```

In this way, we don't have the circular dependency, because the `ReactWideGamutView` will live in the same package as `ReactUIManager` and we can safely cast the view to the right interface and avoid any subclass.
Also, all future subclasses of ReactViewGroup will benefit from implementing the same interface out of the box.

We should check whether there are other React componentes that we need to make them implement the same interface.... 🤔

Reply:
This is also the same approach used for `uimanager.ReactZIndexedViewGroup`: as you can see this interface leaves in the `uimanager` package.

What do you think?

Reply:
That makes sense to me. I'll get it updated. 😄

Reply:
Updated!

> We should check whether there are other React componentes that we need to make them implement the same interface.... 

AFAIK this isn't the case _yet_. Implementing this in ReactViewGroup is all I need to get wide gamut backgrounds working for View.